cargo-workspaces/src/publish.rs

```diff
@@ -54,6 +54,17 @@ pub struct Publish {
     /// Don't remove dev-dependencies while publishing
     #[clap(long)]
     no_remove_dev_deps: bool,
+
+    /// Only print the order in which crates should be published.
+    #[clap(long)]
+    only_print_plan: bool,
+
+    /// Perform basic checks to make sure that crate can be published,
+    /// e.g. that each package has description and license fields.
+    ///
+    /// This is only a heuristic to prevent simple mistakes.
+    #[clap(long)]
+    basic_checks: bool,
```

Comment:
Does `cargo publish` not run these checks?

Reply:
It certainly does, but when publishing a workspace, if, say, 7th package has some field missing, you will only know when you're in the middle of process (when you already bumped a version, created a commit, etc). So this is not very useful for publishing individual packages, but IMHO very useful for publishing a workspace.

Reply:
What do you think of running cargo's `--dry-run` option then instead of writing the checks on our own? That way, we can run our own `--dry-run` for this `publish` subcommand. Here's the related issue https://github.com/pksunkara/cargo-workspaces/issues/4

Reply:
Unfortunately, `--dry-run` doesn't perform these checks, as it aborts before uploading :( 
These checks are performed on the server only.

Reply:
That makes sense. Let's rename the option to `dry_run`, keep the checks and also incorporate cargo's `--dry-run`. Wouldn't completely fix #4, but is a start towards it.

Reply:
Unfortunately, I don't think it would work -- `cargo publish --dry-run` _does_ check that dependencies are published, so `cargo publish --dry-run` would fail for the 2nd crate in the workspace.

And without this functionality, changing the name would probably be misleading.

Reply:
> And without this functionality, changing the name would probably be misleading.

If we are to incorporate everything else from the cargo's dry run other than checking published dependencies, it wouldn't be misleading, right? I know that cargo-release has a dry run option.

I still think we name this option as such and work toward completing it. But, in the meanwhile, we can document it as in progress.

Reply:
@pksunkara that makes sense! I've reworked the implementation:

- Renamed `basic-checks` to `dry-run`.
- Changed `dry-run` behavior to ignore versioning and publishing.
- Added `cargo build` as a step together with `basic_checks` so that we check if package compiles at least. This is not strictly equivalent to `--dry-run` behavior, because AFAIU it will reuse the workspace `Cargo.lock`, while with `--dry-run` a fresh `Cargo.lock` is generated for the package. Also, it obviously doesn't try to use previously published packages, but I guess it's unavoidable.
- With `--dry-run` mode for publish, `--basic-checks` do not make much sense in `plan` subcommand, so I removed it there.
- I also made checking if package is published optional for `plan` subcommand, because the difference in speed is significant and you don't always want to check that.

tl;dr:
- `cargo ws publish --dry-run` seems to have a good enough implementation.
- `cargo ws plan` doesn't try to be too smart and only provides information on publishing plan.